### DISCUSSION TITLE
How to keep upstream settings on my fork?

### DISCUSSION THREAD

**Discussion by SharakPL** [community/community#22039] [New to GitHub]
<p>I have a fork of a project I want to contribute. I want to keep my fork up to date, so I need to <code>git fetch upstream</code>. Though I noticed that every time I clone my fork on a new machine I have to setup upstream again (<code>git remote add upstream https://github.com/ORIGINAL_OWNER/ORIGINAL_REPOSITORY.git</code>). Why is that and how to make this automatic, so that every time I clone my fork upstream is already set up?</p>

**Comment by kingthorin** [comment]
<p>As far as I know it’s just a “that’s how git works” kind of thing. You could make a cli alias that clones and sets the upstream but then you have to set that up on new machines too.</p>

**Comment by SharakPL** [comment, marked as answer]
<p>Actually it can be done automatically, but instead cloning through the console you need to use <a href="https://desktop.github.com/" rel="noopener nofollow ugc">GitHub Desktop</a>. It will setup upstream straight away :slight_smile:</p>
